pkg/jobs/registry.go

```diff
@@ -983,7 +983,9 @@ func (r *Registry) stepThroughStateMachine(
 	status Status,
 	jobErr error,
 ) error {
-	log.Infof(ctx, "job %d: stepping through state %s with error: %v", *job.ID(), status, jobErr)
+	payload := job.Payload()
```

Comment:
nit: I'd use the value directly since the `Payload()` does not return a pointer
jobtype := job.mu.payload.Type().String()

Reply:
but `Payload()` handles the locking, right?

Reply:
I was wary of reaching through `.mu` -- typically, by convention, that is a no-no unless you either locked it or are in a `fooLocked()` method, right?

Reply:
You can wrap it in a Lock/Unlock pair or you can add a function to jobs.go that returns the type of a job protected by Lock/Unlock, probably it will come handy in other places.

Reply:
but `.Payload()` already does the lock/unlock for me? I guess I'm not sure why i'm trying to avoid it? just because it needs to move `payload` to the heap to call `Type()` on it?